File: streaming-server/src/main/java/io/github/hs96wings/streaming_server/video/dto/VideoResDto.java
package io.github.hs96wings.streaming_server.video.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import io.github.hs96wings.streaming_server.video.domain.Video;
import io.github.hs96wings.streaming_server.video.domain.VideoStatus;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class VideoResDto {
    private Long id;
    private String title;
    private String description;
    private String videoPath;
    private String thumbnailPath;
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private LocalDateTime uploadedAt;
    private Long viewCount;

    public VideoResDto(Video video) {
        this.id = video.getId();
        this.title = video.getTitle();
        this.description = video.getDescription();
        this.videoPath = video.getVideoPath();
        this.thumbnailPath = video.getThumbnailPath();
        this.uploadedAt = video.getUploadedAt();
        this.viewCount = video.getViewCount();
    }
}

File: streaming-server/src/main/java/io/github/hs96wings/streaming_server/video/dto/VideoResDto.kt
package io.github.hs96wings.streaming_server.video.dto

import com.fasterxml.jackson.annotation.JsonFormat
import io.github.hs96wings.streaming_server.video.domain.Video
import java.time.LocalDateTime

data class VideoResDto (
    val id: Long,
    val title: String? = null,
    val description: String? = null,
    val videoPath: String? = null,
    val thumbnailPath: String? = null,
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss")
    val uploadedAt: LocalDateTime,
    val viewCount: Long? = null
) {
    constructor(video: Video): this (
        id = video.id ?: throw IllegalStateException("Video ID cannot be null for DTO conversion"),
        title = video.title,
        description = video.description,
        videoPath = video.videoPath,
        thumbnailPath = video.thumbnailPath,
        uploadedAt = video.uploadedAt,
        viewCount = video.viewCount
    )
}